domain/watcher.go

```diff
@@ -75,8 +75,36 @@ func (f *WatcherFactory) NewNamespaceMapperWatcher(
 	), nil
 }
 
+// NewFilterWatcher returns a watcher that combines multiple value
+// watchers into a single watcher.
```

Comment:
The comment isn't quite right - I suspect it was cut and pasted from the old code.
Should we also mention that it returns a notify watcher since we also use string watchers.